kwok/tests/usersignup.go

```diff
@@ -0,0 +1,65 @@
+package kwok
+
+import (
+	"encoding/json"
+	"fmt"
+	"os/exec"
+	"strings"
+)
+
+func PodmanBuildImage() (string, error) {
+	buildCmd := exec.Command("podman", "build", "--build-arg", "CLUSTER_NAME=test", "-t", "my-kwokctl-oc-image", "-f", "../Dockerfile", "../../.")
+	output, err := buildCmd.CombinedOutput()
+	if err != nil {
+		return "", fmt.Errorf("failed to build image: %w", err)
+	}
+
+	fmt.Printf("BuildImage output: %s\n", output)
+	outputString := strings.Split(strings.TrimSpace(string(output)), "\n")
+	imageID := outputString[len(outputString)-1]
+
+	return imageID, nil
+}
+
+func PodmanRunContainer(imageID string) (string, error) {
+	cmd := exec.Command("podman", "run", "-d", "--rm", imageID)
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		return "", fmt.Errorf("failed to start command: %w", err)
+	}
+
+	containerID := strings.TrimSpace(string(output))
+
+	return containerID, nil
+}
+
+func PodmanExecCommand(containerID string, command string) (map[string]int64, error) {
+	cmd := exec.Command("podman", "exec", containerID, "sh", command)
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		return nil, fmt.Errorf("failed to execute command: %w", err)
+	}
+
+	result := strings.TrimSpace(string(output))
+	parsedResult := make(map[string]int64)
+	err = json.Unmarshal([]byte(result), &parsedResult)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse JSON output: %w", err)
+	}
+
+	return parsedResult, nil
+}
+
+func PodmanCleanup(containerID string) error {
+	cmd := exec.Command("podman", "stop", containerID)
+	if err := cmd.Run(); err != nil {
+		return fmt.Errorf("error stopping the container: %w", err)
+	}
+
+	cmd = exec.Command("podman", "rm", "-f", containerID)
+	if err := cmd.Run(); err != nil {
+		return fmt.Errorf("error removing the container: %w", err)
+	}
+
+	return nil
+}
```

Comment:
Are we gonna use this file or is it temporary? it mostly iterates over what's done in my patch

Reply:
I'm planning to use it once it is merged. had that here as a temp way to test things.

Reply:
In the meanwhile you can rebase on top of my change. 
You can do so after this change is merged, saving some time, but it just adds a lot of code to review here which soon won't be needed

Reply:
I will import podman kube play lib from container fixtures but that needs some modification in existing. But that's not a lot.